packages/cli/src/commands/join.ts

```diff
@@ -332,73 +334,179 @@ export async function handleJoin(argv: JoinArgv, packageVersion: string) {
-              api,
-            ];
+
+        const pathItem = paths[path] as Oas3PathItem;
+
+        for (const field of Object.keys(pathItem) as Array<keyof Oas3PathItem>) {
+          if (operationsSet.has(field as Oas3Operations)) {
+            collectPathOperation(pathItem, path, field as Oas3Operations);
           }
-          const { tags, security } = joinedDef.paths[path][operation];
-          if (tags) {
-            joinedDef.paths[path][operation].tags = tags.map((tag: string) =>
-              addPrefix(tag, tagsPrefix)
-            );
-            populateTags({
-              api,
-              apiFilename,
-              tags: formatTags(tags),
-              potentialConflicts,
-              tagsPrefix,
-              componentsPrefix,
-            });
-          } else {
-            joinedDef.paths[path][operation]['tags'] = [
-              addPrefix('other', tagsPrefix || apiFilename),
-            ];
-            populateTags({
-              api,
-              apiFilename,
-              tags: formatTags(['other']),
-              potentialConflicts,
-              tagsPrefix: tagsPrefix || apiFilename,
-              componentsPrefix,
-            });
+          if (field === 'servers') {
+            collectPathServers(pathItem, path);
           }
-          if (!security && openapi.hasOwnProperty('security')) {
-            joinedDef.paths[path][operation]['security'] = addSecurityPrefix(
-              openapi.security,
-              componentsPrefix!
-            );
-          } else if (pathOperation.security) {
-            joinedDef.paths[path][operation].security = addSecurityPrefix(
-              pathOperation.security,
-              componentsPrefix!
-            );
+          if (field === 'parameters') {
+            collectPathParameters(pathItem, path);
+          }
+          if (typeof pathItem[field] === 'string') {
+            collectPathStringFields(pathItem, path, field);
+          }
+        }
+      }
+    }
+
+    function collectPathStringFields(
+      pathItem: Oas3PathItem,
+      path: string | number,
+      field: keyof Oas3PathItem
+    ) {
+      const fieldValue = pathItem[field];
+      if (
+        joinedDef.paths[path].hasOwnProperty(field) &&
+        joinedDef.paths[path][field] !== fieldValue
+      ) {
+        process.stderr.write(yellow(`warning: different ${field} values in ${path}\n`));
+        return;
+      }
+      joinedDef.paths[path][field] = fieldValue;
+    }
+
+    function collectPathServers(pathItem: Oas3PathItem, path: string | number) {
+      if (!pathItem.servers) {
+        return;
+      }
+
+      if (!joinedDef.paths[path].hasOwnProperty('servers')) {
+        joinedDef.paths[path].servers = [];
+      }
+
+      for (const server of pathItem.servers) {
+        let isFoundServer = false;
+        for (const pathServer of joinedDef.paths[path].servers) {
+          if (pathServer.url === server.url) {
+            isFoundServer = true;
```

Comment:
do we need this variable if we anyway terminate the program in case of duplications ?

Reply:
No, we will terminate the program only if two conditions are met. We need this variable to check if we can add the current object to the result